homeassistant/components/v2c/__init__.py

```diff
@@ -31,6 +31,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
+    if coordinator.data.ID and entry.unique_id != coordinator.data.ID:
+        hass.config_entries.async_update_entry(entry, unique_id=coordinator.data.ID)
+
```

Comment:
You can use a minor version migration for this

Reply:
I'm not migrating because users with old firmware will get data.ID == None

Reply:
I must have backwards compatibility with firmwares that don't provide the device ID